### ISSUE TITLE
Zero initial weights

### ISSUE THREAD

**Issue by KinWaiCheuk** [jthickstun/pytorch_musicnet#4]
Inside your Model class, you set the initial weights to be zero

```
        self.linear = torch.nn.Linear(regions*k, m, bias=False).cuda()
        torch.nn.init.constant(self.linear.weight, 0)
```

I wonder what is the reason setting all the weights to zero?
If I start with random weights, the model somehow doesn't work, the accuracy will be very low.

**Comment by jthickstun** [comment]
This is a linear model, so zero-initialization is fine because there are no symmetries in the loss surface that need to be broken. However, a random initialization should also work: e.g., normal mean zero with variance .01. You might need to turn down the learning rate a little bit if you initialize with very large weights.
